### PR TITLE
settlement workers prepare

### PR DESCRIPTION
**Pull request by clD11**
This PR adds functions that can be used to wrap operations and retry them using a configurable retry policy

### CODE REVIEW

File: utils/backoff/backoff.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+			if !IsRetriable(err) {
+				return nil, err
+			}
+
+			if next = retryPolicy.CalculateNextDelay(); next == retrypolicy.Done {
+				return nil, err
+			}
+
+			time.Sleep(next)
```
Comment (husobee) [inline utils/backoff/backoff.go:43]: <-time.After instead?

File: utils/test/random.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ package test
 
 import (
 	"crypto/rand"
+	"math"
```
Comment (husobee) [inline utils/test/random.go:7]: :(

Comment (clD11) [reply utils/test/random.go:7]: its import for constant math.MaxInt32

File: utils/backoff/backoff.go (hunk 1 above)
Comment (clD11) [reply utils/backoff/backoff.go:43]: will that not allocate a new channel/timer which will most likely not be used everytime this is called then need GC etc considering high throughput for this. Syscall are the same in both cases although thread state slightly different. A timer with reset would be slight more idiomatic

**Review by husobee — APPROVED** [review]

**Pull request merged**